### PR TITLE
Initial support for dark mode

### LINKED ISSUE #6268: Support for dark mode
Check that Openshift dark mode is propagated to Kiali application and make it compatible in those cases that dark mode is not propagated correctly.

### PR DESCRIPTION
**Pull request by josunect**
Fixes #6268 

Requires OSSMC changes: https://github.com/kiali/openshift-servicemesh-plugin/pull/185

Notes: 

- How to persist dark mode in Kiali

### CODE REVIEW

File: frontend/src/components/Nav/Masthead/UserDropdown.tsx [hunk 1]
```diff
@@ -93,6 +98,16 @@ class UserDropdownComponent extends React.Component<UserProps, UserState> {
     }
   };
 
+  handleTheme = () => {
+    if (this.state.theme === themes[0]) {
+      this.setState({ theme: themes[1] });
+      document.getElementsByTagName('html')[0].classList.add(darkmode);
```
Comment (ferhoyos) [inline frontend/src/components/Nav/Masthead/UserDropdown.tsx:104]: HTML can be retrieved by `document.documentElement` (https://developer.mozilla.org/en-US/docs/Web/API/Document/documentElement)

File: frontend/src/components/Nav/Masthead/UserDropdown.tsx [hunk 2]
```diff
@@ -116,9 +131,17 @@ class UserDropdownComponent extends React.Component<UserProps, UserState> {
...
-        Logout
-      </DropdownItem>
+      <>
+        {' '}
+        {canLogout && (
+          <DropdownItem key={'user_logout_option'} onClick={this.handleLogout} isDisabled={!canLogout}>
+            Logout
+          </DropdownItem>
+        )}
+        <DropdownItem key={'them_update'} onClick={this.handleTheme}>
```
Comment (ferhoyos) [inline frontend/src/components/Nav/Masthead/UserDropdown.tsx:141]: typo in `key={'theme_update'}`

File: frontend/src/components/Nav/Masthead/UserDropdown.tsx [hunk 3]
```diff
@@ -116,9 +131,17 @@ class UserDropdownComponent extends React.Component<UserProps, UserState> {
...
+          {this.state.theme === themes[0] ? themes[1] : themes[0]} theme
```
Comment (ferhoyos) [inline frontend/src/components/Nav/Masthead/UserDropdown.tsx:142]: I think the first letter better in capital like 'Logout'

**Review by ferhoyos — CHANGES_REQUESTED** [review]

File: frontend/src/components/Envoy/EnvoyDetails.tsx [hunk 4]
```diff
@@ -23,6 +23,7 @@ import { SummaryTableBuilder } from './tables/BaseTable';
 import { Namespace } from 'types/Namespace';
 import { kialiStyle } from 'styles/StyleUtils';
 import AceEditor from 'react-ace';
+import 'ace-builds/src-noconflict/theme-twilight';
```
Comment (ferhoyos) [inline frontend/src/components/Envoy/EnvoyDetails.tsx:26]: Why theme-twilight is imported here but not theme-eclipse? Is this import needed here or we have to add theme-eclipse too?

Comment (josunect) [reply frontend/src/components/Envoy/EnvoyDetails.tsx:26]: When I've tried the theme was not working, so I've added the import statement. 
I thought the eclipse was included as a default.

File: frontend/package.json [hunk 5]
```diff
@@ -111,6 +111,7 @@
     "@types/lodash-es": "4.17.x",
     "@types/node": "12.12.14",
     "@types/react": "^17.0.1",
+    "@types/react-datepicker": "^4.15.0",
     "@types/react-dom": "^17.0.1",
```
Comment (aljesusg) [inline frontend/package.json:115]: Why react-date-picker?

Comment (ferhoyos) [reply frontend/package.json:115]: When I was adapting react-datepicker to dark theme I had this warning message from the editor:
```
Could not find a declaration file for module 'react-datepicker'. '/home/fhoyosle/kiali/kiali_sources/kiali/frontend/node_modules/react-datepicker/dist/index.js' implicitly has an 'any' type.
Try `npm i --save-dev @types/react-datepicker` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-datepicker'; ts(7016)`
```
For this reason I have added the types declaration of react-datepicker library to package.json

**Review by ferhoyos — APPROVED** [review]

**Review by ferhoyos — APPROVED** [review]
LGTM

**Pull request merged**